src/Time/Units.hs

```diff
@@ -100,6 +114,41 @@ instance KnownSymbol (ShowUnit unit) => Show (Time unit) where
                                                 n -> '/' : show n
                       in numeratorStr ++ denominatorStr ++ symbolVal (Proxy @(ShowUnit unit))
 
+----------------------------------------------------------------------------
+-- Creation helpers
+----------------------------------------------------------------------------
+
+sec :: Natural -> Second
+sec n = Time (n % 1)
```

Comment:
You repeat `Time (n % 1)` pattern a lot. I think you can create helper function:

```haskell
time :: Natural -> Time unit
```

and then implement every function just as `sec = time` `ms = time`, etc. Every function is just a specialization of `time` function. `time` can be exported in case people want to create their own time units.